tests/cnf/core/network/metallb/tests/metallb-nodeSelector.go

```diff
@@ -0,0 +1,319 @@
+
+				frrPod0, frrPod1 := setupTestCase(ipAddressPool1, ipAddressPool2)
+
+				By("Creating a BGPAdvertisement with nodeSelector for bgpPeer1")
+				workerNode0Label := []metav1.LabelSelector{
+					{MatchLabels: map[string]string{tsparams.LabelHostName: workerNodeList[0].Definition.Name}},
+				}
+
+				workerNode1Label := []metav1.LabelSelector{
+					{MatchLabels: map[string]string{tsparams.LabelHostName: workerNodeList[1].Definition.Name}},
+				}
+
+				By("Creating a BGPAdvertisement with the nodeSelector to bgppeer1")
+				setupBgpAdvertisementWithNodeSelector(bgpAdvertisementName1, noAdvertise, 100,
+					[]string{bgpPeerName1}, ipAddressPool1, workerNode0Label, true)
+
+				setupBgpAdvertisementWithNodeSelector(bgpAdvertisementName2, customCommunity, 200,
+					[]string{bgpPeerName2}, ipAddressPool2, workerNode1Label, true)
+
+				verifyBGPConnectivityAndPrefixes(frrPod0, frrPod1, nodeAddrList, addressPool1, addressPool2)
+
+				By("Validate Local Preference from Frr node0")
+				err = frr.ValidateLocalPref(frrPod0, 100, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Fail to validate local preference")
+
+				By("Validate Local Preference from Frr node1")
+				err = frr.ValidateLocalPref(frrPod1, 200, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Fail to validate local preference")
+
+				By("Validate BGP Community exists with the node selector")
+				output, err := frr.GetBGPCommunityStatus(frrPod0, noAdvertise, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
+				Expect(len(output.Routes)).To(Equal(1))
+
+				By("Validate BGP Community does not exist without the node selector")
+				output, err = frr.GetBGPCommunityStatus(frrPod1, customCommunity, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
+				Expect(len(output.Routes)).To(Equal(1))
+			})
+	})
+})
+
+func createIPAddressPool(name string, ipPrefix []string) *metallb.IPAddressPoolBuilder {
+	ipAddressPool, err := metallb.NewIPAddressPoolBuilder(
+		APIClient,
+		name,
+		NetConfig.MlbOperatorNamespace,
+		[]string{fmt.Sprintf("%s-%s", ipPrefix[0], ipPrefix[1])}).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create IPAddressPool")
+
+	return ipAddressPool
+}
+
+func createBGPPeer(name, peerIP string) error {
+	_, err := metallb.NewBPGPeerBuilder(APIClient, name, NetConfig.MlbOperatorNamespace,
+		peerIP, tsparams.LocalBGPASN, tsparams.LocalBGPASN).WithPassword(tsparams.BGPPassword).Create()
+
+	return err
+}
+
+func setupTestCase(ipAddressPool1, ipAddressPool2 *metallb.IPAddressPoolBuilder) (*pod.Builder, *pod.Builder) {
+	By("Creating two MetalLB service")
+
+	setupMetalLbService("service-1", netparam.IPV4Family, ipAddressPool1, "Cluster")
+	setupMetalLbService("service-2", netparam.IPV4Family, ipAddressPool2, "Cluster")
```

Comment:
is that expected that you create 2 the same services with different name in case single IP addressPool?

Reply:
There are two different ipaddresspools

Reply:
I'm talking about line [112 ](https://github.com/openshift-kni/eco-gotests/pull/347/files#diff-4852af88224ac4d493264a975a5f0d100a32127b9bfc0500ec2ae4c874339518R112
)
`frrPod0, frrPod1 := setupTestCase(ipAddressPool, ipAddressPool, frrk8sPods)`

Reply:
?

Reply:
Yes there is two services on a single IPaddressPool in the original test case but lets talk to make sure I understand your question.